### ISSUE TITLE
AUPR-Error and AUPR-Success

### ISSUE THREAD

**Issue by zzzyzh** [cognizant-ai-labs/red-paper#2]
Hi,
For ood detection, do we generally need to use AUPR-Error or AUPR-Success?
If I now calculate the ood score `score` of the model, and the `gt` of ood, should the score be calculated using the method `average_precision_score(gt, score)` provided by `sklearn` be AUPR-Success or AUPR-Error?
This question has been bothering me for a long time and I look forward to your reply!

**Comment by VsonicV** [comment]
@zzzyzh Hi, the metric "AUPR-Error" and "AUPR-Success" in the paper are used for error (misclassification made by the original base classifier) detections, so the "error" and "success" here means misclassification and correct classification (by the original base classifier), respectively. This is different in the case of OOD detection, which we should use AUPR-OOD instead. Here, the AUPR-OOD means the area under the Precision-Recall (AUPR) Curve when treating OOD samples as positive class during the detection. The `average_precision_score` function is used to calculate the AP (Average Precision) score, not AUPR score. If you want to get AUPR-OOD, here is an example code you can use:
```
from sklearn.metrics import precision_recall_curve, auc
precision, recall, thresholds = precision_recall_curve(ood_true, detection_score)
auc(recall, precision)
```
The `ood_true` contains the binary labels indicate whether a sample is OOD, and the `detection_score` contains the score used for OOD detection, e.g., the RED-variance used in the paper.

Sorry for this late reply. Hope it helps!
